File: NapakalakiRuby/lib/Monster.rb
#encoding: UTF-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module NapakalakiGame
  class Monster
    attr_accessor :name, :combatLevel, :prize, :bc, :levelChangeAgainstCultistPlayer
    def initialize(name, combatLevel, prize, bc)
      @name = name
      @combatLevel = combatLevel
      @prize = prize
      @bc = bc
      @levelChangeAgainstCultistPlayer = 0
    end
    
    def initialize(name, combatLevel, prize, bc, lv)
      @name = name
      @combatLevel = combatLevel
      @prize = prize
      @bv = bc
      @levelChangeAgainstCultistPlayer = lv
    end
    
    def getName
      @name
    end
    
    def getCombatLevel
      @combatLevel
    end
    
    def getPrize
      @prize
    end
    
    def getBadConsequence
      @bc
    end
    
    def getCombatLevelAgainstCultistPlayer
      return getCombatLevel() + @levelChangeAgainstCultistPlayer
    end
    
    def to_s
      "Nombre: #{@name} Nivel: #{@combatLevel} Precio: #{@prize} Bad Consequence: #{@bc}"
    end
    
  end
end

File: Napakalaki/src/NapakalakiGame/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author afro
 */
public class Monster {
    private String name;
    private int combatLevel;   
    private Prize price;
    private BadConsequence bc;
    private int levelChangeAgainstCultistPlayer;
    
    public Monster(String name, int combatLevel){
        this.name = name;
        this.combatLevel = combatLevel;
        levelChangeAgainstCultistPlayer = 0;
    }

    public Monster(String name, int combatLevel, Prize price, BadConsequence bc) {
        this.name = name;
        this.combatLevel = combatLevel;
        this.price = price;
        this.bc = bc;
        levelChangeAgainstCultistPlayer = 0;
    }
    
    public Monster(String name, int combatLevel, Prize price, BadConsequence bc, int lev) {
        this.name = name;
        this.combatLevel = combatLevel;
        this.price = price;
        this.bc = bc;
        levelChangeAgainstCultistPlayer = lev;
    }
    
    public void setName(String name) {
        this.name = name;
    }

    public void setCombatLevel(int combatLevel) {
        this.combatLevel = combatLevel;
    }

    public void setPrice(Prize price) {
        this.price = price;
    }

    public void setBc(BadConsequence bc) {
        this.bc = bc;
    }

    public String getName() {
        return name;
    }

    public int getCombatLevel() {
        return combatLevel;
    }
    
    @Override
    public String toString(){        
        return "Nombre = " +name+"Combat level="+ Integer.toString(combatLevel)+"Precio="+price+"Consecuence="+bc;
    } 

    public Prize getPrice() {
        return price;
    }

    public BadConsequence getBc() {
        return bc;
    } 
    
    public int getLevelsGained(){
        return price.getLevels();
    }
    
    public int getTreasuresGained(){
        return price.getTreasures();
    }

    int getSpecialValue() {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }
    
    public int getCombatLevelAgainstCultist(){
        return getCombatLevel() + levelChangeAgainstCultistPlayer;
    }
}
